### DISCUSSION TITLE
How to use progress

### DISCUSSION THREAD

**Discussion by nampoinaandriamalala** [norkunas/youtube-dl-php#170] [Q&A] [answered]
Hi all!

I tested "Download video" and it's run perfectly.
Now , i want check every 5 second the progress of the download and i created file php with "Download progress" and i ran it but there is nothing showed.

**Comment by norkunas** [comment, marked as answer]
Hi, did you mean you have used `onProgress` callback?

**Reply by nampoinaandriamalala** [reply]
Thanks for reply. I want it onProgress callback

**Reply by norkunas** [reply]
Are you using it in a http context or a console context?

**Reply by nampoinaandriamalala** [reply]
I want check it first on console.

**Reply by norkunas** [reply]
So it should work then, it's covered by tests

**Reply by nampoinaandriamalala** [reply]
I think that i don't know how to use it. 

![image](https://user-images.githubusercontent.com/38759562/147105806-0e85df14-9554-4983-a026-2ee3558bd764.png)

When i run php progres.php i have nothing

**Reply by norkunas** [reply]
Usage looks good, not sure than. If you could paste your full script I could try to run myself locally

**Reply by nampoinaandriamalala** [reply]
TY! I tested on console. It's run and i have progress. 


![image](https://user-images.githubusercontent.com/38759562/147209092-92f57de3-70db-480e-9abf-00081c31d032.png)

![image](https://user-images.githubusercontent.com/38759562/147209192-165efa83-d3be-46db-90d1-249b4ab3eeb5.png)

**Reply by mijnnaamisramon** [reply]
When I try this, with the exact same code, the output only shows up at the end.... not at intervals. So I get all the output at the same time. I checked the json that is created during downloading but there are no values in there I can use to poll it and show it (like making a progress bar). If someone knows how to make this, that would be awesome!

**Reply by norkunas** [reply]
@mijnnaamisramon if you run it via web that's expected, your download process must be detached.
to properly provide progress to the UI you could use [SSE](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events) like https://mercure.rocks/ or web sockets.

**Comment by mijnnaamisramon** [comment]
Thank you!
